**Antonio Foglia, af3152**

Notebook: https://github.com/Antonio-Foglia/ASTRC3105

In [1]:
import numpy as np
import astropy as ap
from astropy import constants as const
from astropy  import units as u
import latexify

In [2]:
identifiers = {
    "M_star": "M",
    "R_star": "R",
    "M_sun": "M_s",
    "R_sun": "R_s",
    "M_earth": "M_e",
    "R_earth": "R_e",
    "hour": "h",
    "day": "d",
}
G = const.G
pi = np.pi
M_sun = const.M_sun
R_sun = const.R_sun
R_earth = const.R_earth
day = u.day
hour = u.hour
AU = const.au

# Q1

In [3]:
@latexify.function(identifiers=identifiers, use_math_symbols=True)
def a(P, M_star):
    return ((P**2 * G * M_star / (4 * pi**2))**(1/3))
a

In [4]:
a(100*day,0.6*M_sun).decompose()

<Quantity 5.31998259e+10 m>

In [5]:
@latexify.function(identifiers=identifiers, use_math_symbols=True)
def Prob(P, M_star, R_star):
    return 0.005*(R_star / R_sun) * (a(P, M_star) / (1*AU))**(-1)
Prob

In [20]:
Prob(100 * day, 0.6*M_sun, 0.51*R_sun).decompose()

<Quantity 0.0071706>

## b)

In [7]:
@latexify.function(identifiers=identifiers, use_math_symbols=True)
def t(P, M_star, R_star):
    return 13*(M_star / M_sun)**(-1/2) * (a(P, M_star) / (1*AU))**(1/2) * (R_star / R_sun) * hour
t

In [8]:
t(100 * day, 0.6*M_sun, 0.51*R_sun).to(u.hour)

<Quantity 5.10423079 h>

## c)

In [9]:
Rp = 3389.5 * 1000 * u.m #source https://en.wikipedia.org/wiki/Mars

In [10]:
(Rp/(0.51*R_sun)).decompose()

<Quantity 0.00955308>

# Q2

In [11]:
M = 0.69*M_sun
R = 0.64*R_sun

In [12]:
@latexify.function(identifiers=identifiers, use_math_symbols=True, reduce_assignments=True)
def R_p(minimum_L):
    flux_change = (1 - minimum_L)/1
    # print(f"DL/L*: {flux_change}\n")
    
    Rp = R * np.sqrt(flux_change)
    #print(f"Rp: {Rp.to(u.R_earth)}\n")
    return Rp
R_p

In [13]:
@latexify.function(identifiers=identifiers, use_math_symbols=True, reduce_assignments=True)
def r(tau):
    a = (1*AU) * (tau/13)**2 * (M/M_sun) * (R/R_sun)**(-2) * (1/hour)**(2)
    #print(f"a: {a.to(u.AU)}\n")
    return a
r

## a) & b)

### Kepler-62b

In [14]:
print(f'a) R_p = {R_p(0.9996).to(u.R_earth).round(3)}\nb) a = {r(2.9 * hour).to(u.AU).round(3)}')

a) R_p = 1.396 earthRad
b) a = 0.084 AU


### Kepler-62c

In [15]:
print(f'a) R_p = {R_p(0.9999).to(u.R_earth).round(3)}\nb) a = {r(3.5 * hour).to(u.AU).round(3)}')

a) R_p = 0.698 earthRad
b) a = 0.122 AU


### Kepler-62d

In [16]:
print(f'a) R_p = {R_p(0.9990).to(u.R_earth).round(3)}\nb) a = {r(3.5 * hour).to(u.AU).round(3)}')

a) R_p = 2.208 earthRad
b) a = 0.122 AU


### Kepler-62e

In [17]:
print(f'a) R_p = {R_p(0.9993).to(u.R_earth).round(3)}\nb) a = {r(7.1 * hour).to(u.AU).round(3)}')

a) R_p = 1.847 earthRad
b) a = 0.502 AU


### Kepler-62f

In [18]:
print(f'a) R_p = {R_p(0.9995).to(u.R_earth).round(3)}\nb) a = {r(8 * hour).to(u.AU).round(3)}')

a) R_p = 1.561 earthRad
b) a = 0.638 AU


## c)

In [19]:
a(18.2 * day, M).to(u.AU)

<Quantity 0.11965523 AU>

The above is quite similar to what I got in the calculation in part *b)*: $0.122$ AU.

We were making the assumption that the orbits were perfectly circular, which is often not the case. Moreover we were assuming $i \approx 90^{\circ}$ which means that the planet is orbiting on the same plane as the observer, which is also unlikely.